x/programs/cmd/simulator/cmd/plan.go

```diff
@@ -202,7 +201,11 @@ func (c *runCmd) RunStep(ctx context.Context, db *state.SimpleMutable) (*Respons
 	// map all transactions to their step_N identifier
 	txID, found := resp.getTxID()
 	if found {
-		c.programIDStrMap[fmt.Sprintf("step_%d", index)] = txID
+		id, err := ids.FromString(txID)
+		if err != nil {
+			return nil, err
+		}
+		c.programIDStrMap[fmt.Sprintf("step_%d", index)] = id
```

Comment:
Why not just use the index instead of the string, the `step_` is redundant here. You could probably also use a sparse `[]id.ID` and fill in a bunch `nil` for all the other outputs. None of this needs to be performant.

Reply:
Replaced the type of the key of the map to `int`. Using a list would also have worked but would have required a bit more of complexity.